packages/core/content-releases/server/src/bootstrap.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import type { LoadedStrapi } from '@strapi/types';
+import type { Entity as StrapiEntity } from '@strapi/types';
+import { RELEASE_ACTION_MODEL_UID } from './constants';
+
+const { features } = require('@strapi/strapi/dist/utils/ee');
+
+export const bootstrap = async ({ strapi }: { strapi: LoadedStrapi }) => {
+  if (
+    features.isEnabled('cms-content-releases') &&
+    strapi.features.future.isEnabled('contentReleases')
+  ) {
+    strapi.db.lifecycles.subscribe({
+      afterDelete(event) {
+        // @ts-expect-error TODO: lifecycles types looks like are not 100% finished
+        const { model, result } = event;
+
+        // @ts-expect-error TODO: lifecycles types looks like are not 100% finished
+        if (model.kind === 'collectionType' && model.options.draftAndPublish) {
+          const { id } = result;
+
+          strapi.db.query(RELEASE_ACTION_MODEL_UID).deleteMany({
+            where: {
+              target_type: model.uid,
+              target_id: id,
+            },
+          });
+        }
+      },
+
+      /**
+       * deleteMany hook doesn't return the deleted entries ids
+       * so we need to fetch them before deleting the entries to save the ids on our state
+       */
+      async beforeDeleteMany(event) {
+        const { model, params } = event;
+
+        // @ts-expect-error TODO: lifecycles types looks like are not 100% finished
+        if (model.kind === 'collectionType' && model.options.draftAndPublish) {
+          const { where } = params;
+
+          const entriesToDelete = await strapi.db
+            .query(model.uid)
+            .findMany({ select: ['id'], where });
+
+          event.state.entriesToDelete = entriesToDelete;
```

Comment:
Why can't you delete the release action here instead? Do you think the delete event can fail?

Reply:
I think it's an option 🤔. If, for some reason, the delete event fails, you'll be left with the entries not deleted, but the actions on the release would be deleted.

Do you think that's very unlikely and not worth implementing this logic?

Reply:
I'm a bit confused why you even need this beforeDeleteMany and why you need to set the event.state.entriesToDelete. Couldn't you just query the entries to delete in the afterDeleteMany?

Reply:
You can't query the entries on afterDeleteMany because they are deleted, for that reason I make the query before, and then use those ids to delete the release actions.

[Here](https://docs.strapi.io/dev-docs/backend-customization/models#hook-event-object) we mention that we can use state to share info between before and after events